### PR TITLE
packages input for turnbeat

### PR DESCRIPTION
**Pull request by bikrant**
Currently supports only RPM system

### CODE REVIEW

File: inputs/packages/packages.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package packages
+
+/* an empty input, useful as a starting point for a new input */
```
Comment (johann8384) [inline inputs/packages/packages.go:3]: Let's go ahead and get a good description, author, copyright block here.

**Comment by johann8384** [comment]
LGTM, just change the description and I'll merge it.

**Comment by johann8384** [comment]
Can you post an example of the output in JSON format?

**Comment by johann8384** [comment]
I'm thinking an object that looks something like:

```
packages {
  installed: [...],
  available: [...]
}
```

This code appears to just list installed, and I'm happy to merge it with just that but we should make a card in Trello to enhance it, unless you go and do it now.

It looks like individual record format is great, based on the code it is:

```
{
  name: 'foo',
  version: '4.0.2-1',
  arch: 'noarch'
}
```

**Comment by gilbertgong** [comment]
I'll probably base the executor plugin on portions of this w/ ideas from Jonathan's.  I like the simplicity here but I'll bring in an idea or two from Jonathan's.

**Comment by johann8384** [comment]
It looks like this runs the command and _then_ parses the output. I think the executor I wrote streams the output to the channel as it happens, line by line.

Am I reading this properly? If so, I think I want the stream behavior for the general purpose executor, or at the very least to have that as an option.

**Comment by gilbertgong** [comment]
Well, it reads the streaming input, and uses the input to create a bunch of RPMPackage objects, creates an event out of the objects, and then writes the event to the channel.  I envision the general purpose executor to read the input, convert it to a basic event (where all it does it populate the "message" field), and then write the event to the channel.

**Comment by johann8384** [comment]
You can work off a PR by pulling the PR into your branch without merging the PR, although in this case, merging the PR makes sense anyway.
https://gist.github.com/piscisaureus/3342247

**Comment by gilbertgong** [comment]
Streaming to the output channel is very easy to code, but do we "want" that
behavior?  I thought you always wanted events to be sent through the
channels..  Doesn't that break a bunch of expectations and conventions if
we write a stream to the channel instead of an event?  Like don't filters
and outputs expect to receive events?

Gilbert

**Pull request merged**